### PR TITLE
Read settings from environment variables instead of a config file.

### PR DESCRIPTION
**Pull request by maurizi**
Connects to #66

### CODE REVIEW

**Comment by jwalgran** [comment]
We should remove the `-config` flag from the upstart template https://github.com/maurizi/otm-cloud/blob/850de9f62675b7c30935d4028f679687b164650d/deployment/ansible/roles/otm.ecoservice/templates/eco-upstart.conf.j2#L14

**Comment by jwalgran** [comment]
The "Running the ecoservice" section of the README should be updated to enumerate the environment variables required instead of explaining the config file.

**Comment by jwalgran** [comment]
`config.gcfg.template` should be removed

File: main.go [hunk 1]
```diff
@@ -1,7 +1,6 @@
 package main
 
 import (
-	"code.google.com/p/gcfg"
```
Comment (hectcastro) [inline main.go]: Aside from the `README` and the configuration file example, there are also references to the configuration file inside of the `Makefile`.

File: ecorest/config/main.go [hunk 2]
```diff
@@ -2,19 +2,49 @@ package config
...
+
 type Config struct {
 	Database eco.DBInfo
-	Data     struct {
-		Path string
-	}
-	Server struct {
-		Host string
-		Port string
+	Data     data
```
Comment (hectcastro) [inline ecorest/config/main.go]: Do we even need the `Server` and `Data` structs above? Seems like they could just be fields in `Config`.

File: main.go (hunk 1 above)
Comment (maurizi) [inline main.go]: Those were removed in https://github.com/maurizi/otm-ecoservice/commit/e69f34f1ab6b408823ff3f98ef890a731c2945f4

File: ecorest/config/main.go (hunk 2 above)
Comment (maurizi) [inline ecorest/config/main.go]: Yeah, that simplifies things a bit.  Done in https://github.com/OpenTreeMap/otm-ecoservice/commit/ae4773fd4770d5908da22e997a2f49ec4e33e8d5

**Comment by maurizi** [comment]
@jwalgran I removed the remaining references to the config and fixed a few other things, this is ready for another look.

File: ecorest/config/main.go [hunk 3]
```diff
@@ -2,19 +2,46 @@ package config
 
 import (
 	"github.com/OpenTreeMap/otm-ecoservice/eco"
+	"os"
 )
 
+type server struct {
```
Comment (hectcastro) [inline ecorest/config/main.go]: Doesn't seem like these are being used any more, right?

Comment (maurizi) [inline ecorest/config/main.go]: Oh woops. Gone in https://github.com/maurizi/otm-ecoservice/commit/1d614340e63070434c403f751fcaceeb0f91e30d

**Comment by jwalgran** [comment]
One tiny remnant of the config file is left in .gitignore

https://github.com/maurizi/otm-ecoservice/blob/1d614340e63070434c403f751fcaceeb0f91e30d/.gitignore#L13

**Comment by jwalgran** [comment]
I ran `vagrant provision app` and `vagrant reload app` and had a running and fully functional ecoservice with no config file.

:+1:

**Pull request merged**